src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -782,6 +784,30 @@ void ObjectMgr::LoadCreatureTemplateAddons()
             }
         }
 
+        if (standState >= MAX_UNIT_STAND_STATE)
+        {
+            TC_LOG_ERROR("sql.sql", "Creature (Entry: %u) has invalid unit stand state (%u) defined in `creature_addon`. Truncated to 0", entry, standState);
+            standState = 0;
+        }
+        creatureAddon.bytes1 |= uint32(uint32(standState) << (UNIT_BYTES_1_OFFSET_STAND_STATE * 8));
+
+        if (AnimTier(animTier) >= AnimTier::Max)
+        {
+            TC_LOG_ERROR("sql.sql", "Creature (Entry: %u) has invalid animation tier (%u) defined in `creature_addon`. Truncated to 0", entry, standState);
+            animTier = 0;
+        }
+        creatureAddon.bytes1 |= uint32(uint32(animTier) << (UNIT_BYTES_1_OFFSET_ANIM_TIER * 8));
+
+        if (sheathState >= MAX_SHEATH_STATE)
+        {
+            TC_LOG_ERROR("sql.sql", "Creature (Entry: %u) has invalid sheath state (%u) defined in `creature_addon`. Truncated to 0", entry, standState);
+            sheathState = 0;
+        }
+        creatureAddon.bytes2 |= uint32(uint32(sheathState) << (UNIT_BYTES_2_OFFSET_SHEATH_STATE * 8));
```

Comment:
Split them in CreatureAddon structure too (they get deconstructed into single fields anyway when loading addon in Creature class)

Reply:
kk. now while looking at it. what to do with visbility flags? (untrackable etc). are these meant to be handled by the core as well?

Reply:
For now just add columns for fields we load in Creature::LoadCreaturesAddon()

Reply:
I think they should all be handled by the core. At least the untrackable and invisible (https://github.com/TrinityCore/TrinityCore/pull/28448) flags I have only seen applied in the presence of auras

Reply:
for now I do as requested and keep the flags in a db column until the core features are in place